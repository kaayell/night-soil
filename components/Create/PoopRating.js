import React from 'react'
import {FormLabel} from 'react-native-elements'
import {View} from 'react-native'
import {POPPINS} from '../StyleGuide/fonts'
import StarRating from "react-native-star-rating";

export const PoopRating = ({selected, onRatingChange}) => {
  const labelText = [
    'Kill Me',
    'Shouldn\'t have had spicy wings',
    'Average',
    'Quick and easy',
    'Did I even poop?']
  return <View style={{flex: 1}}>
    <FormLabel
      labelStyle={{color: 'white', fontSize: 20, textAlign: 'center', paddingBottom: 10}}
      fontFamily={POPPINS}
    >
      {labelText[selected - 1]}
    </FormLabel>
    <StarRating
      maxStars={5}
      disabled={false}
      rating={selected}
      fullStarColor={"FFCC00"}
      selectedStar={onRatingChange}
      containerStyle={{flex: 1, justifyContent: 'center', alignItems: 'center'}}
    />
  </View>
}

export default PoopRating