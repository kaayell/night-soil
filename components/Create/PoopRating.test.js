import React from "react";
import PoopRating from "./PoopRating";
import {shallow} from "enzyme";
import StarRating from "react-native-star-rating";

describe('PoopRating', () => {
  it('should render', () => {
    let wrapper = shallow(<PoopRating/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('should invoke callback on selected star', () => {
    const mockSelected = jest.fn()
    let wrapper = shallow(<PoopRating onRatingChange={mockSelected}/>)
    wrapper.find(StarRating).props().selectedStar()
    expect(mockSelected).toHaveBeenCalled()
  });
});