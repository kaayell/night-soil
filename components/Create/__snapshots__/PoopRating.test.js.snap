// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoopRating should render 1`] = `
<View>
  <FormLabel
    fontFamily="poppins-regular"
    labelStyle={
      Object {
        "color": "#FAFAFA",
        "fontSize": 15,
        "paddingBottom": 5,
        "textAlign": "center",
      }
    }
  />
  <StarRating
    buttonStyle={Object {}}
    containerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
    disabled={false}
    emptyStar="star-o"
    emptyStarColor="gray"
    fullStar="star"
    fullStarColor="#FFCC00"
    halfStar="star-half-o"
    halfStarEnabled={false}
    iconSet="FontAwesome"
    maxStars={5}
    rating={0}
    reversed={false}
    selectedStar={[Function]}
    starSize={40}
    starStyle={Object {}}
  />
</View>
`;
